cloudevents/working-drafts/dds-protocol-binding.md

```diff
@@ -0,0 +1,265 @@
+- Filtering and Matching: Publishers use key values to categorize data samples, and
+subscribers use key-based filters to specify which data samples they are
+interested in. This reduces the amount of data that is sent over the wire in a
+DDS-based system.
+- Reliability and Durability: These Quality of Service (QoS) settings control
+how DDS handles reliable message delivery of data samples with the same key, and
+when to overwrite locally available data samples with the same key.
+
+While it is not necessary to set the `datakey` field for a DDS CloudEvent message,
+setting this value will enable many of the more powerful features of the DDS
+protocol.
+
+### 3.2. Binary Content Mode
+
+The _binary_ content mode accommodates any shape of event data, and allows for
+efficient transfer and without transcoding effort.
+
+#### 3.2.1. Content Type
+
+For the _binary_ mode, the header `content-type` property MUST be mapped
+directly to the CloudEvents `datacontenttype` attribute and the
+`datacontentencoding` attribute MUST be set to `binary`.
+
+#### 3.2.2. Event Data Encoding
+
+The [`data`](#21-data) byte-sequence MUST be used as the body ("Data" field)
+of the DDS message.
+
+In binary mode, the DDS representation of a CloudEvent with no `data` is a
+DDS message with no body. Transmission such a DDS message is allowable.
+
+#### 3.2.3. Metadata Headers
+
+All [CloudEvents][ce] attributes and [CloudEvent Attributes Extensions]
+(../primer.md#cloudevent-extension-attributes)
+with exception of `data` MUST be individually mapped to and from the Header
+fields in the DDS message.
+
+### 3.3. Structured Content Mode
+
+The _structured_ content mode keeps event metadata and data together in the
+payload, allowing simple forwarding of the same event across multiple routing
+hops, and across multiple protocols.
+
+#### 3.3.1. Content-Type
+
+If present, the DDS message header property `content-type` MUST be set to the
+media type of an [event format](#14-event-formats).
+
+Example for the [JSON format][json-format]:
+
+```text
+content-type: cloudevent/json
+```
+Example for the [DDS event format][dds-message-format]:
+
+```text
+content-type: application/cloudevent+dds
+```
+
+For both of the above cases, the `content-type` property MUST be mapped
+directly to the CloudEvents `datacontenttype` attribute and the
+`datacontentencoding` attribute MUST be set to either `text` or `json`
+depending on the type of structured data being transmitted.
+
+
+#### 3.3.2. Event Data Encoding
+
+The chosen [event format](#14-event-formats) defines how all attributes, and
+`data`, are represented.
+
+The event metadata and data are then rendered in accordance with the
+[event format](#14-event-formats) specification and the resulting data becomes
+the DDS application [data](#21-data) section.
+
+Similar to binary mode, the DDS representation of a structured CloudEvent with
+no `data` is a DDS message with no body, and transmission of such a message
```

Comment:
`is a DDS message with no body`... is this true? Wouldn't a structured CE still have a DDS message+body(`data`) but just missing the `data` attribute of the CE structure? Meaning, the DDS message's `data` would have the CE JSON but in that JSON the `data` attribute would be empty/missing. Or am I reading this incorrectly?

Reply:
No, you are reading it correctly, my wording is incorrect. I changed the wording to the following:

The DDS representation of a structured CloudEvent with no `data` is a message
that contains all of the CE attributes but has an empty body. Transmission of
such a message is allowable.

Does this seem reasonable?